src/common/oclengine.cpp

```diff
@@ -103,12 +103,14 @@ void OCLEngine::InitOCL()
     for (int i = 0; i < deviceCount; i++) {
         // a context is like a "runtime link" to the device and platform;
         // i.e. communication is possible
-        std::shared_ptr<OCLDeviceContext> devCntxt = std::make_shared<OCLDeviceContext>(devPlatVec[i], all_devices[i]);
+        std::shared_ptr<OCLDeviceContext> devCntxt = std::make_shared<OCLDeviceContext>(*(devPlatVec[i]), all_devices[i]);
```

Comment:
OCLDeviceContext doesn't it shouldn't take a smart pointer parameter? That seems wierd and inconsistent to me.